docs/application/web/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,445 @@
+
+- Start and stop data flow to a pipeline branch
+
+  You can [use Valve API to stop and resume data flow within a pipeline branch](#start-and-stop-data-flow-to-a-pipeline-branch)
+
+- Write custom data filters
+
+  You can [use CustomFilter API to write custom data processing routines in JS](#write-custom-data-filters)
+
+- Use saved models
+
+  You can use [`tensor_filter` element to read models trained with popular machine learning frameworks](#use-saved-models)
+
+## Prerequisites
+To access files, camera or recorder using the Pipeline API (in [mobile](../../api/latest/device_api/mobile/tizen/ml_pipeline.html), [wearable](../../api/latest/device_api/wearable/tizen/ml_pipeline.html), and [tv](../../api/latest/device_api/tv/tizen/ml_pipeline.html) applications) the application has to define proper privileges in its `config.xml`:
+
+```xml
+<!-- for accessing internal storage only -->
+<tizen:privilege name="http://tizen.org/privilege/mediastorage"/>
+<!-- for accessing external storage only -->
+<tizen:privilege name="http://tizen.org/privilege/externalstorage"/>
+<!-- for accessing camera -->
+<tizen:privilege name="http://tizen.org/privilege/camera"/>
+<!-- for accessing recorder -->
+<tizen:privilege name="http://tizen.org/privilege/recorder"/>
+```
+
+As these are [privacy-related privileges](../../tutorials/sec-privileges.md) the application has to [request proper permissions using the PPM API](../security/privacy-related-permissions.md) (in [mobile](../../api/latest/device_api/mobile/tizen/ppm.html) and [wearable](../../api/latest/device_api/wearable/tizen/ppm.html) applications).
+
+
+## Create and dispose machine learning inference pipelines
+NNStreamer is a plugin for [GStreamer](https://gstreamer.freedesktop.org), adding tensor data types and enabling use of machine learning inference models in pipelines.
+You can use some of the standard GStreamer and NNStreamer elements (also called _nodes_ throughout this guide) in your applications.
+
+NNStreamer pipelines are created from string descriptions, for example:
+```javascript
+'appsrc ! other/tensor,dimension=(string)2:1:1:1,type=(string)int8,framerate=(fraction)0/1 ! tensor_filter framework=custom-easy model=my-custom-filter ! tensor_sink'
+```
+Exclamation marks `!` separate pipeline nodes.
+You can learn how to define pipeline descriptions from [GStreamer documentation](https://gstreamer.freedesktop.org/documentation).
+
+To create a machine learning inference pipeline:
+
+1. Describe it as a string:
+   ```javascript
+   var pipelineDescription = 'videotestsrc num-buffers=3 ' +
+                             '! video/x-raw,width=20,height=15,format=BGRA ' +
+                             '! tensor_converter ' +
+                             '! fakesink';
+   ```
+
+2. Call `tizen.ml.pipeline.createPipeline()`:
+   ```javascript
+   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
+   ```
```

Comment:
Global: 
Please give one line space after a sentence is followed by code Like below:
What is **it** please can you replace it with actual subject.
```suggestion
1. Describe it as a string:

   ```javascript
   var pipelineDescription = 'videotestsrc num-buffers=3 ' +
                             '! video/x-raw,width=20,height=15,format=BGRA ' +
                             '! tensor_converter ' +
                             '! fakesink';
   ```

2. Call `tizen.ml.pipeline.createPipeline()`:

   ```javascript
   var pipeline = tizen.ml.pipeline.createPipeline(pipelineDescription);
   ```